--- a/platformio/debug/process/gdb.py
+++ b/platformio/debug/process/gdb.py
@@ -181,3 +181,11 @@
     def stderr_data_received(self, data):
         super().stderr_data_received(data)
         self._handle_error(data)
+        if helpers.is_gdbmi_mode():
+            if is_bytes(data):
+                data = data.decode()
+            data = data.strip()
+            if data:
+                self.stdout_data_received(
+                    helpers.escape_gdbmi_stream("&", data + "\n")
+                )
+            return
+        super().stderr_data_received(data)
+        self._handle_error(data)
